tensorflow_io/json/README.md

```diff
@@ -0,0 +1,207 @@
+# TensorFlow i/o JSON Datasets
+
+[JSON (JavaScript Object Notation)](https://www.json.org/) is an open-standard
+file format that uses human-readable text to transmit data objects consisting
+of key–value pairs and array data types (or any other serializable value). JSON
+format is widely used in machine learning and data science.
+
+TensorFlow i/o JSON support mainly focuses on the **record** array in JSON, here
+is an example:
+
+```json
+[
+  {
+    "floatfeature": 1.1,
+    "integerfeature": 2,
+    "floatlabel": 1.2
+  },
+  {
+    "floatfeature": 2.1,
+    "integerfeature": 3,
+    "floatlabel": 1.1  
+  }
+]
+```
+
+## Custom Ops
+
+There are 2 different custom operators (
+[Eager Execution](https://www.tensorflow.org/guide/eager) support only)
+supported by JSON Datasets: `list_json_columns` and `read_json`.
+
+### `list_json_columns`
+
+`list_json_columns` is used to fetch all the column names of the JSON format
+files. Here is an example:
+
+```python
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow as tf
+# Enable Eager Execution.
+tf.enable_eager_execution()
+import tensorflow_io.json as json_io
+
+# Assume `example.json` is a JSON format file that contains the content of the
+# example above.
+filename = "example.json"
+cols = json_io.list_json_columns(filename)
+```
+
+### `read_json`
+
+`read_json` can be used to read a specified column of the JSON file into a
```

Comment:
does it only read one single column or one or more columns?

Reply:
Good catch :) 
Currently, it only read one single column, the reason for that is that I just followed the Python API of avro dataset, but now I am thinking about maybe it will be better to `zip` the tensors internally, in this way, user can read more columns and also can specify columns order.

cc @yongtang , Do you know the reason that the python API of avro dataset only allows the user to read a single column?

Reply:
@jiachengxu @BryanCutler Avro was done with single column for no particular reason, other than I was hoping tf.io could be implemented in a more column-friendly way.

As was mentioned in other threads, at the moment dataset is not truly column-oriented (more row-oriented) so it is not easy to split columns once all columns has been read into a dataset.

On the other hand, it is fairly easy to go the other route: combine multiple single-column dataset into one dataset. It could be done with `zip`, or we could easily expand an API to allow doing that (see my example in the comments of this PR).

As mentioned it is fine if the dataset is closer to destination (tf.keras), but for many data sources, additional feature engineering is needed and it is ready hard to use dataset if it is row-oriented.